modules/gui/src/display/vpDisplayOpenCV.cpp

```diff
@@ -1254,19 +1254,40 @@ void vpDisplayOpenCV::displayCross(const vpImagePoint &ip, unsigned int size, co
   \param color : Line color.
   \param thickness : Line thickness.
 */
-void vpDisplayOpenCV::displayDotLine(const vpImagePoint &ip1, const vpImagePoint &ip2, const vpColor &color,
+void vpDisplayOpenCV::displayDotLine(const vpImagePoint &ip1_, const vpImagePoint &ip2_, const vpColor &color,
                                      unsigned int thickness)
 {
   if (m_displayHasBeenInitialized) {
+    vpImagePoint ip1 = ip1_;
+    vpImagePoint ip2 = ip2_;
+
     double size = 10. * m_scale;
     double length = sqrt(vpMath::sqr(ip2.get_i() - ip1.get_i()) + vpMath::sqr(ip2.get_j() - ip1.get_j()));
-    double deltaj = size / length * (ip2.get_j() - ip1.get_j());
+    bool vertical_line = (int)ip2.get_j() == (int)ip1.get_j();
+    if (vertical_line) {
+      if (ip2.get_i() < ip1.get_i()) {
+        std::swap(ip1, ip2);
+      }
+    } else if (ip2.get_j() < ip1.get_j()) {
+      std::swap(ip1, ip2);
+    }
+
+    double diff_j = vertical_line ? 1 : ip2.get_j() - ip1.get_j();
+    double deltaj = size / length * diff_j;
     double deltai = size / length * (ip2.get_i() - ip1.get_i());
-    double slope = (ip2.get_i() - ip1.get_i()) / (ip2.get_j() - ip1.get_j());
+    double slope = (ip2.get_i() - ip1.get_i()) / diff_j;
     double orig = ip1.get_i() - slope * ip1.get_j();
-    for (unsigned int j = (unsigned int)ip1.get_j(); j < ip2.get_j(); j += (unsigned int)(2 * deltaj)) {
-      double i = slope * j + orig;
-      displayLine(vpImagePoint(i, j), vpImagePoint(i + deltai, j + deltaj), color, thickness);
+
+    if (vertical_line) {
+      for (unsigned int i = (unsigned int)ip1.get_i(); i < ip2.get_i(); i += (unsigned int)(2 * deltai)) {
+        double j = ip1.get_j();
+        displayLine(vpImagePoint(i, j), vpImagePoint(i + deltai, j), color, thickness);
+      }
+    } else {
+      for (unsigned int j = (unsigned int)ip1.get_j(); j < ip2.get_j(); j += (unsigned int)(2 * deltaj)) {
```

Comment:
Here `ip1.get_j()` must be < `ip2.get_j()`. So we need to swap  when it is not the case.